NavigateU/NavigateU/RegistrationScreen/RegisterModelProtocol.swift

```diff
@@ -0,0 +1,20 @@
+//
+//  RegisterModelProtocol.swift
+//  NavigateU
+//
+//  Created by Faki Doosuur Doris on 06.05.2024.
+//
+
+import Foundation
+import Combine
+
+protocol RegistrationProtocol: ObservableObject where ObjectWillChangePublisher.Output == Void {
+    associatedtype State
+    var state: State { get }
+
+    func register(firstName: String, lastName: String, email: String, password: String, countryOfOrigin: String, cityOfResidence: String)
+}
+
+protocol RegViewModel: RegistrationProtocol {
+    var stateDidChangeForReg: ObservableObjectPublisher { get }
+}
```

Comment:
we shouldn't add protocols for every ViewModel, that one from my example is enough for every viewmodel